### PR TITLE
Added new tests for two consecutive adjoint runs

### PR DESCRIPTION
**Pull request by sabakhshi**
## Purpose
In the development effort for SST, it was discovered that Tapenade was causing changes to the solutions states after an adjoint run due to a bug. To ensure that this was not a systematic problem throughout the code and to catch future instances of this issues appearing, I added two new adjoint tests. The first runs the adjoint twice back to back to ensure that the results didn't change. The second runs the adjoint twice back to back and checks to make sure that the states have not changed between the runs.

## Expected time until merged
2-3 weeks

## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [x] Maintenance update
- [ ] Other (please describe)

## Testing
The new feature is a new test.

## Checklist

- [ ] I have run `flake8` and `black` to make sure the Python code adheres to PEP-8 and is consistently formatted
- [ ] I have formatted the Fortran code with `fprettify` or C/C++ code with `clang-format` as applicable
- [x] I have run unit and regression tests which pass locally with my changes
- [x] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

File: tests/reg_tests/reg_test_utils.py [hunk 1]
```diff
@@ -13,6 +13,24 @@ def assert_adjoint_sens_allclose(handler, CFDSolver, ap, evalFuncs=None, **kwarg
     handler.root_print("Eval Functions Sens:")
     handler.root_add_dict("Eval Functions Sens:", funcsSens, rtol=rtol, atol=atol)
 
+def assert_adjoint2_sens_allclose(handler, CFDSolver, ap, evalFuncs=None, **kwargs):
+    rtol, atol = getTol(**kwargs)
+    funcsSens = {}
+    CFDSolver.evalFunctionsSens(ap, funcsSens, evalFuncs=None)
+    CFDSolver.evalFunctionsSens(ap, funcsSens, evalFuncs=None)
+    handler.root_print("Eval Functions Sens 2:")
+    handler.root_add_dict("Eval Functions Sens 2:", funcsSens, rtol=rtol, atol=atol)
```
Comment (marcomangano) [inline tests/reg_tests/reg_test_utils.py:22]: Shouldn't we compare the second run against the values of the first run `Eval Functions Sens`? I don't see the point of adding this if the point of the test is to make sure that the adjoint converges to the same result (within a tolerance)

File: tests/reg_tests/reg_test_utils.py [hunk 2]
```diff
@@ -43,7 +61,7 @@ def assert_residuals_allclose(handler, CFDSolver, ap, **kwargs):
     res /= totalR0
     handler.root_print("Norm of residual")
     handler.par_add_norm("Norm of residual", res, rtol=rtol, atol=atol)
-
+
```
Comment (marcomangano) [inline tests/reg_tests/reg_test_utils.py:64]: This is probably causing the black and flake8 error

File: tests/reg_tests/refs/adjoint_euler_scalar_jst_tut_wing.json [hunk 3]
```diff
@@ -1,86 +1,87 @@
 {
-    "Eval Functions Sens:": {
+    "Adjoint States": 0.0,
+    "Eval Functions Sens 2:": {
```
Comment (marcomangano) [inline tests/reg_tests/refs/adjoint_euler_scalar_jst_tut_wing.json:3]: As I mention below, I don't think we should add this, ideally the only thing added to the `.json` files is the `Adjoint States` entry

**Review by marcomangano — CHANGES_REQUESTED** [review]
See my comments below, I don't think we should add separate reference results for the second sensitivity analysis, am I missing anything?

File: tests/reg_tests/refs/adjoint_euler_scalar_jst_tut_wing.json [hunk 4]
```diff
@@ -1,86 +1,87 @@
 {
+    "Adjoint States": 0.0,
```
Comment (marcomangano) [inline tests/reg_tests/refs/adjoint_euler_scalar_jst_tut_wing.json:2]: This entry should be the only one changing in the `.json` files. There is no point changing the reference values for this PR, especially if the training is done on a local machine.

**Review by marcomangano — CHANGES_REQUESTED** [review]
The added tests look good, I only think the `.json` files need to be cleaned up - see my comment below

**Comment by marcomangano** [comment]
Also note that the black and flake8 tests are failing, please fix them. Remember that we currently use `black==23.1.0` and custom `flake8` settings

**Comment by sabakhshi** [comment]
Everything should be addressed. Re-requesting your review.

**Review by marcomangano — APPROVED** [review]
Nice cleanup, thanks!

**Review by lamkina — APPROVED** [review]

**Pull request merged**